native/src/main/java/io/ballerina/stdlib/http/api/HttpUtil.java

```diff
@@ -1344,6 +1344,14 @@ public static void populatePoolingConfig(BMap poolRecord, PoolConfiguration pool
                 maxActiveStreamsPerConnection == -1 ? Integer.MAX_VALUE : validateConfig(
                         maxActiveStreamsPerConnection,
                         HttpConstants.CONNECTION_POOLING_MAX_ACTIVE_STREAMS_PER_CONNECTION.getValue()));
+        double minEvictableIdleTime =
+                ((BDecimal) poolRecord.get(HttpConstants.CONNECTION_POOLING_EVICTABLE_IDLE_TIME)).floatValue();
+        poolConfiguration.setMinEvictableIdleTime(minEvictableIdleTime < 0 ? 0 : (long) minEvictableIdleTime * 1000);
```

Comment:
Usually if a configuration value has a negative value like -1 ect. the configuration will be considered as disabled. Can we apply that principle here rather that setting the value as 0 ?

Reply:
I think this is because, here we are setting this value to the Apache connection pool which is a third-party library, and there "0" is considered as the default.

Reply:
This is a patch backported from 2201.8.x. I dont think its good to change the configuration behaviour.
And also, this is a value we forward to apache generic object pool. so disabling such config does not make sense IMO

Reply:
This is a patch backported from 2201.8.x. I dont think its good to change the configuration behaviour.
And also, this is a value we forward to apache generic object pool. so disabling such config does not make sense IMO

Reply:
Ack. But, I think our original approach should have used the disabling config when the value is a negative value. But, since you are passing this config to Apache generic pool object it might be doing something similar from within. But, its better to confirm the behaviour when the value is negative or zero.